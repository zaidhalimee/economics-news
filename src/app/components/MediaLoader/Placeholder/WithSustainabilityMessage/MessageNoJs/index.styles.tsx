import pixelsToRem from '#app/utilities/pixelsToRem';
import { css, Theme } from '@emotion/react';

const WIDTH = 250;

const styles = {
  container: ({ palette, spacings, mq }: Theme) =>
    css({
      textAlign: 'start',
      position: 'absolute',
      color: palette.WHITE,
      padding: `${spacings.DOUBLE}rem`,
      background: `rgba(34, 34, 34, 0.75)`,
      height: '100%',
      width: '100%',
      zIndex: '1',
      [mq.GROUP_1_MIN_WIDTH]: {
        textAlign: 'end',
        left: `calc(100% - ${pixelsToRem(WIDTH)}rem)`,
        width: `${pixelsToRem(WIDTH)}rem`,
        height: 'unset',
      },
    }),
  message: ({ palette }: Theme) =>
    css({
      color: palette.WHITE,
      margin: '0',
    }),
};

export default styles;
