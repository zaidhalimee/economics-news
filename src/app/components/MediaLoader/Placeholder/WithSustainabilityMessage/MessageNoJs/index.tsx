/** @jsx jsx */
import { jsx } from '@emotion/react';
import Text from '#app/components/Text';
import styles from './index.styles';

type Props = {
  noJsMessage: string;
};

const MessageNoJs = ({ noJsMessage }: Props) => {
  return (
    <noscript>
      <div css={styles.container}>
        <Text css={styles.message} as="p" size="longPrimer">
          {noJsMessage}
        </Text>
      </div>
    </noscript>
  );
};

export default MessageNoJs;
