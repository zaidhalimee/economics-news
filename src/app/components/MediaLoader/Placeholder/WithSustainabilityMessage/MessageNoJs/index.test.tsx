import React from 'react';
import { render } from '../../../../react-testing-library-with-providers';
import MessageNoJs from '.';

describe('Message No JS', () => {
  afterEach(() => {
    jest.resetModules();
  });

  it(`should render a 'Help reduce your power consumption' message`, () => {
    const { container } = render(
      <MessageNoJs noJsMessage="Help reduce your power and data usage by not playing video content." />,
      {
        service: 'mundo',
      },
    );
    expect(container.querySelector('noscript')).toBeInTheDocument();
  });
});
