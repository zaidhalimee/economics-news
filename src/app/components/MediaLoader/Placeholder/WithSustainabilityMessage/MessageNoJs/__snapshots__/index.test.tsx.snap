// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message No JS should render a 'Help reduce your power consumption' message 1`] = `
<div>
  <noscript />
</div>
`;
